version: '3.8'

services:

  frontend:
    image: sermalenk/skypro-front:lesson-34
    ports:
      - "80:80"
    depends_on:
      - db
      - api
    networks:
      - avito_front_network

  api:
    image: kpaveliev/skypro-c06-cw06-todolist:${GITHUB_REF_NAME}-${GITHUB_RUN_ID}
    depends_on:
      migrations:
        condition: service_completed_successfully
      db:
        condition: service_healthy
    ports:
      - "8000:8000"
    volumes:
      - ./.env:/code/.env
    networks:
      - avito_front_network
      - avito_db_network


  migrations:
    image: kpaveliev/skypro-c06-cw06-todolist:${GITHUB_REF_NAME}-${GITHUB_RUN_ID}
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - ./.env:/code/.env
    networks:
      - avito_db_network
    command: >
      sh -c "python manage.py migrate"


  db:
    image: postgres:12.4-alpine
    volumes:
      - postgres_avito:/var/lib/postgresql/data
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_NAME}
    restart: always
    ports:
      - "${DB_PORT}:${DB_PORT}"
    networks:
      - avito_db_network
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ${DB_USER} -d ${DB_NAME}" ]
      interval: 5s
      timeout: 5s
      retries: 5


volumes:
  postgres_avito:


networks:
  avito_db_network:
  avito_front_network:
